src/iceberg/manifest_entry.h

```diff
@@ -0,0 +1,286 @@
+  /// Value field id: 130
+  /// Map from column id to upper bound in the column serialized as binary.
+  /// Each value must be greater than or equal to all non-null, non-NaN values in the
+  /// column for the file.
+  std::map<int32_t, std::vector<uint8_t>> upper_bounds;
+  /// Field id: 131
+  /// Implementation-specific key metadata for encryption
+  std::optional<std::vector<uint8_t>> key_metadata;
+  /// Field id: 132
+  /// Element Field id: 133
+  /// Split offsets for the data file. For example, all row group offsets in a Parquet
+  /// file. Must be sorted ascending.
+  std::vector<int64_t> split_offsets;
+  /// Field id: 135
+  /// Element Field id: 136
+  /// Field ids used to determine row equality in equality delete files. Required when
+  /// content=2 and should be null otherwise. Fields with ids listed in this column must
+  /// be present in the delete file.
+  std::vector<int32_t> equality_ids;
+  /// Field id: 140
+  /// ID representing sort order for this file
+  ///
+  /// If sort order ID is missing or unknown, then the order is assumed to be unsorted.
+  /// Only data files and equality delete files should be written with a non-null order
+  /// id. Position deletes are required to be sorted by file and position, not a table
+  /// order, and should set sort order id to null. Readers must ignore sort order id for
+  /// position delete files.
+  std::optional<int32_t> sort_order_id;
+  /// This field is not included in spec, so it is not serialized into the manifest file.
+  /// It is just store in memory representation used in process.
+  int32_t partition_spec_id;
+  /// Field id: 142
+  /// The _row_id for the first row in the data file.
+  ///
+  /// Reference:
+  /// - [First Row ID
+  /// Inheritance](https://github.com/apache/iceberg/blob/main/format/spec.md#first-row-id-inheritance)
+  std::optional<int64_t> first_row_id;
+  /// Field id: 143
+  /// Fully qualified location (URI with FS scheme) of a data file that all deletes
+  /// reference.
+  ///
+  /// Position delete metadata can use referenced_data_file when all deletes tracked by
+  /// the entry are in a single data file. Setting the referenced file is required for
+  /// deletion vectors.
+  std::optional<std::string> referenced_data_file;
+  /// Field id: 144
+  /// The offset in the file where the content starts.
+  ///
+  /// The content_offset and content_size_in_bytes fields are used to reference a specific
+  /// blob for direct access to a deletion vector. For deletion vectors, these values are
+  /// required and must exactly match the offset and length stored in the Puffin footer
+  /// for the deletion vector blob.
+  std::optional<int64_t> content_offset;
+  /// Field id: 145
+  /// The length of a referenced content stored in the file; required if content_offset is
+  /// present
+  std::optional<int64_t> content_size_in_bytes;
+
+  inline static const SchemaField kContent = MakeRequiredField<IntType>(134, "content");
```

Comment:
Should we support adding `doc` to `SchemaField`? In the Java impl, the doc strings are serialized with the schema in the Avro file (manifest list file and manifest file). We can add it later though.

Am I right? @Fokko